### PR TITLE
Refactor#3622

### LINKED ISSUE #3622: Centralize tests in Modin
Right now, Modin tests are split between the top-level `modin/test` and in the directories (e.g. `modin/pandas/test`). I would like to centralize the test location to make it easier to run the test suite with `pytest`.

### PR DESCRIPTION
**Pull request by teaVeloper**
## What do these changes do?
Working on Issue #3622
Move Tests for modin.pandas into directory modin/test/pandas and changing all references either in

imports
conftest.py
file readers
documentation
Github Actions

not resolving issue completely, as same needs to be done for other tests (I will do, after this PR is accepted, or shall I work on the whole issue before openin PR?).

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3622  
- [x] tests added and passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

**Review by devin-petersohn — COMMENTED** [review]
@teaVeloper The structure looks great. We have some flaky tests at the moment, so I am rerunning CI to make sure everything is working as intended.

@YarShev can we get TeamCity updated to reflect this new test structure?

**Comment by YarShev** [comment]
> @YarShev can we get TeamCity updated to reflect this new test structure?

Sure, but first we will need to be agreed on the changes. As soon as we are okay with the changes, we will update TeamCity to run tests. If tests pass in TeamCity too, we'll merge the PR. Does that makes sense?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -147,7 +147,7 @@ jobs:
           conda list
       - run: sudo apt update && sudo apt install -y libhdf5-dev
       - name: Api tests
-        run: python -m pytest modin/pandas/test/test_api.py
+        run: python -m pytest modin/test/pandas/test_api.py
```
Comment (YarShev) [inline .github/workflows/ci.yml:150]: Do we want to put all test directories in one place (modin/config/test, modin/core/execution/dispatching/factories/test, modin/experimental/core/execution/native/implementations/omnisci_on_native/test, etc.)?

Comment (devin-petersohn) [reply .github/workflows/ci.yml:150]: @YarShev I think that will be best, yes.

Comment (teaVeloper) [reply .github/workflows/ci.yml:150]: Hi, so i guess I should continue refactoring with all folders (i wanted to check if its fine with that one folder, but I guess thats not the best workflow, especially for this issue) and after I will update the PR and then it makes sense to check for the modification with the CI and test pipeline?

Comment (devin-petersohn) [reply .github/workflows/ci.yml:150]: @teaVeloper Yes, I think that would be good. It looks great so far!

**Comment by YarShev** [comment]
@teaVeloper, @devin-petersohn, what is the status here?

**Comment by teaVeloper** [comment]
> @teaVeloper, @devin-petersohn, what is the status here?

@YarShev 
Im sorry for having been silent that long, as I have been pretty busy. I will be able to get all done in the next couple days and update the pull request then regarding.

**Comment by YarShev** [comment]
@teaVeloper, great, thanks! Feel free to ask any questions.

**Comment by devin-petersohn** [comment]
@teaVeloper thanks so much for the work on this so far!

The conflicts look pretty complicated. @RehanSD led a major refactor, do you mind if he takes a look and tries to help resolve the conflicts here?

**Comment by teaVeloper** [comment]
Hi @devin-petersohn  thanks for feedback. 

Sure I am happy to get assistance by @RehanSD with this issues!
Also I did rebase on origin/master before continuing my edits, I hope though I did not mess anything up with resolving the conflicts locally.

**Comment by RehanSD** [comment]
Hi @teaVeloper! Just wanted to check in to see how best we could collaborate! We could either use the [Modin slack](slack.modin.org) to pair program and resolve the conflicts, or if you'd prefer, I can always resolve them locally (on my laptop) on your branch, and push to this PR! Please let me know which one you'd prefer! Looking forward to collaborating!

**Comment by teaVeloper** [comment]
Hi @RehanSD thanks for reaching out! I would prefer to pair program and using slack (but I need an invitation I guess?) if that's fine for you? Ill send an email to your berkeley.edu mail address for detailed communication.

**Comment by YarShev** [comment]
@RehanSD, @teaVeloper, what is the status here?

**Comment by RehanSD** [comment]
Hi @YarShev! I think that this PR is stale - unless @teaVeloper has interest in taking it up again, I'd recommend closing for now, and attempting a refactor in a new PR?

**Pull request closed without merging**